.github/workflows/end-to-end-test.yaml

```diff
@@ -0,0 +1,55 @@
+name: end-to-end-test
+on: push
+jobs:
+  end-to-end:
+    runs-on: ubuntu-20.04
+    steps:
+    - name: checkout the repo
+      uses: actions/checkout@v3
+    - name: build the worker-operator
+      uses: docker/build-push-action@v2
+      with:
+        tags: worker-operator:latest
+        push: false
+    - name: determine node-ip
+      run:  sh .github/workflows/scripts/kind-nodeip.sh
+    - name: create kind clusters
+      run: sh .github/workflows/scripts/kind.sh
+    - name: clone the e2e repo
+      uses: actions/checkout@v3 
+      with:
+        repository: saivenkatesh1909/kubeslice-e2e
+        token: ${{ secrets.PAT }}
+        path: kubeslice-e2e
+    - name: copy the kubeconfig
+      run:  cp -r /home/runner/.kube/config /home/runner/work/worker-operator/worker-operator/kubeslice-e2e/assets/kubeconfig/config
+    - name: tag replace to the e2e suite worker-operator
+      run: sudo sed -i 's/        Tag:.*/        Tag:latest/'  kubeslice-e2e/profile/profile_kind_two.yaml  &&   sudo sed -i 's/        Image:.*/        Image:worker-operator/'  kubeslice-e2e/profile/profile_kind_two.yaml    
+    - name: check the change
+      run: cat kubeslice-e2e/profile/profile_kind_two.yaml
+    - name: run the e2e using the worker-operator image
+      uses: docker/build-push-action@v2
+      with:
+        context: ./kubeslice-e2e
+        file: kubeslice-e2e/test-prod.Dockerfile
+        build-args: PROFILE=profile_kind_two
+        tags: end-to-end:latest
+        push: false
+    - name: Docker Extract
+      uses: shrink/actions-docker-extract@v1
+      id: extract
```

Comment:
`actions-docker-extract` is to extract files from an image. Here we need to do `docker run` with mounted volumes instead